packages/auth/src/stateMachine/types.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *	 http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+import { Invocation } from './invocation';
+import { MachineState } from './machineState';
+
+/**
+ * Base type for a Machine's context
+ */
+export type MachineContext = object;
```

Comment:
Maybe `unknown`?
```suggestion
export type MachineContext = Record<string, unknown>;
```

This would be more type-safe as users are required to assert/cast the value type. It's annoying, but safer to some extent. Plus it will work with union of other property types: [playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAsghgYwBYEsB2EDCB7NwIAewUAvFAEoQLYBOAJgDwDOwN6A5gDRQCuaA1mmwB3NAD4A3ACgpoSFADK2ALYR4ydFlz4ipWIlQYceQsQBkUAN5SotqADNs2AFxQWbNFxt2ARnBquPk4ANhBwaFIAvjJ0VMH+0NRoLFBJOsCuSqrqhlomRNJSaaYAdH40QA)

Reply:
Agreed, `object` can be inferred as Array, Function or Object in JS

Reply:
Fixed!